File: related_scripts/verification/step_E/debug_movmean.py
import sys
from pathlib import Path
import numpy as np
sys.path.insert(0, str(Path(__file__).resolve().parents[3] / "src"))
from caz.gridding import _movmean_2d, _movstd_2d

SCRIPT_DIR = Path(__file__).parent

x = np.loadtxt(SCRIPT_DIR / "debug_x.csv", delimiter=",").reshape(1, -1)
y6_mat = np.loadtxt(SCRIPT_DIR / "debug_movmean_k6.csv", delimiter=",").reshape(1, -1)
y5_mat = np.loadtxt(SCRIPT_DIR / "debug_movmean_k5.csv", delimiter=",").reshape(1, -1)
x30 = np.loadtxt(SCRIPT_DIR / "debug_x30.csv", delimiter=",").reshape(1, -1)
y30_mat = np.loadtxt(SCRIPT_DIR / "debug_movmean_k30.csv", delimiter=",").reshape(1, -1)

# Test k=6
y6_py = _movmean_2d(x, 6)
print("k=6 comparison:")
print(f"  MATLAB: {y6_mat[0,:5]}")
print(f"  Python: {y6_py[0,:5]}")
print(f"  max|diff|: {np.max(np.abs(y6_py - y6_mat)):.6e}")

# Test k=5
y5_py = _movmean_2d(x, 5)
print(f"\nk=5 comparison:")
print(f"  MATLAB: {y5_mat[0,:5]}")
print(f"  Python: {y5_py[0,:5]}")
print(f"  max|diff|: {np.max(np.abs(y5_py - y5_mat)):.6e}")

# Test k=30
y30_py = _movmean_2d(x30, 30)
print(f"\nk=30 comparison:")
print(f"  MATLAB first 5: {y30_mat[0,:5]}")
print(f"  Python first 5: {y30_py[0,:5]}")
print(f"  MATLAB last 5: {y30_mat[0,-5:]}")
print(f"  Python last 5: {y30_py[0,-5:]}")
print(f"  max|diff|: {np.max(np.abs(y30_py - y30_mat)):.6e}")

# Show where biggest diff is
diffs = np.abs(y30_py[0] - y30_mat[0])
idx = np.argmax(diffs)
print(f"  Biggest diff at index {idx}: py={y30_py[0,idx]:.6f}, mat={y30_mat[0,idx]:.6f}")

File: related_scripts/verification/step_E/debug_movmean.m
% Debug movmean window behavior
rng(42);
x = randn(1, 20);
writematrix(x, 'debug_x.csv');

% movmean with k=6 (even)
y6 = movmean(x, 6, 2, 'omitnan');
writematrix(y6, 'debug_movmean_k6.csv');

% movmean with k=5 (odd)
y5 = movmean(x, 5, 2, 'omitnan');
writematrix(y5, 'debug_movmean_k5.csv');

% movmean with k=30 on larger data
x30 = randn(1, 200);
writematrix(x30, 'debug_x30.csv');
y30 = movmean(x30, 30, 2, 'omitnan');
writematrix(y30, 'debug_movmean_k30.csv');

% movstd with k=30
ys30 = movstd(x30, 30, 0, 2, 'omitnan');
writematrix(ys30, 'debug_movstd_k30.csv');

fprintf('x = '); disp(x);
fprintf('movmean(x,6) = '); disp(y6);
fprintf('movmean(x,5) = '); disp(y5);
fprintf('First 5 of movmean(x30,30) = '); disp(y30(1:5));
fprintf('Last 5 of movmean(x30,30) = '); disp(y30(end-4:end));
